From b1951a36be4ff14005d6f771cd498dac05b29636 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sat, 16 Dec 2023 13:03:46 -0500
Subject: [PATCH 07/43] init: Add set_custom_settings function

Comes with the following settings overrides: - SET_SCREEN_OFF_TIMEOUT - SET_SLEEP_TIMEOUT - SET_POWER_ALWAYS_ON - SET_STAY_ON_WHILE_PLUGGED_IN - FORCE_BLUETOOTH_SERVICE (enable,disable) - FORCE_DISABLE_ALL_RADIOS (forces airplane mode, removing all radios from being toggled, also removing all QS tiles for radios)
---
 init.sh | 54 ++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 54 insertions(+)

diff --git a/init.sh b/init.sh
index b1d8d68..b70e906 100644
--- a/init.sh
+++ b/init.sh
@@ -1177,6 +1177,59 @@ function set_package_opts()
 	
 }
 
+function set_custom_settings()
+{
+	# Set generic device settings
+	# Example: SET_SCREEN_OFF_TIMEOUT=1800000 # 30 minutes 
+	# 		   SET_SLEEP_TIMEOUT=86400000 # 1 day
+	#
+	for c in `cat /proc/cmdline`; do
+        case $c in
+            *=*)
+                eval $c
+                if [ -z "$1" ]; then
+                    case $c in
+						SET_SCREEN_OFF_TIMEOUT=*)
+							# Set screen off timeout
+							# options: integer in milliseconds
+							settings put system screen_off_timeout "$SET_SCREEN_OFF_TIMEOUT"
+							;;
+						SET_SLEEP_TIMEOUT=*)
+							# Set screen sleep timeout
+							# options: integer in milliseconds
+							settings put system sleep_timeout "$SET_SLEEP_TIMEOUT"
+							;;
+						SET_POWER_ALWAYS_ON=*)
+							# Set power always on
+							# options: true or false
+							svc power stayon "$SET_POWER_ALWAYS_ON"
+							;;
+						SET_STAY_ON_WHILE_PLUGGED_IN=*)
+							# Set stay on while plugged in
+							# options: true or false
+							settings put global stay_on_while_plugged_in "$SET_STAY_ON_WHILE_PLUGGED_IN"
+							;;
+						FORCE_BLUETOOTH_SERVICE=*)
+							# Set force bluetooth service state
+							# options: enable, disable
+							pm "$FORCE_BLUETOOTH_SERVICE" com.android.bluetooth
+							svc bluetooth "$FORCE_BLUETOOTH_SERVICE"
+							;;
+						FORCE_DISABLE_ALL_RADIOS=1)
+							# Set force disable all radios
+							settings put global airplane_mode_radios cell,wifi,bluetooth,nfc,wimax
+							settings put global airplane_mode_toggleable_radios ""
+							settings put secure sysui_qs_tiles "rotation,caffeine,$(settings get secure sysui_qs_tiles)"
+							cmd connectivity airplane-mode enable
+							;;
+						
+                    esac
+                fi
+                ;;
+        esac
+    done
+	
+}
 
 
 function do_init()
@@ -1272,6 +1325,7 @@ function do_bootcomplete()
 		pm disable com.blissos.updater
 	fi
 
+	set_custom_settings
 	set_custom_package_perms
 	set_package_opts
 
-- 
2.34.1

